### PR TITLE
[DOC] Clarify Kafka Connect dependencies when configuring MM2

### PR DESCRIPTION
**Pull request by PaulRMellor**
Signed-off-by: prmellor <pmellor@redhat.com>

**Documentation**

Minor updates to the MM2 configuration procedure in the _Using Guide_ to clarify the dependencies between the MM2 and Kafka Connect versions used.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [ ] Make sure all tests pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

File: documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc [hunk 1]
```diff
@@ -54,6 +54,8 @@ This procedure shows a configuration that uses TLS encryption and authentication
 ** link:{BookURLDeploying}#deploying-kafka-cluster-str[Kafka cluster^]
 * Source and target Kafka clusters must be available
 
+WARNING: To avoid conflicts between dependencies, MirrorMaker 2.0 and the underlying Kafka Connect deployment that supports it must be from the same Kafka version.
```
Comment (scholzj) [inline documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc:57]: I do not think this really should be a warning. There is no real way to do it. So I think we just need to clarify it to not get too many questions about it.

File: documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc [hunk 2]
```diff
@@ -178,9 +180,9 @@ spec:
             name: aws-creds
             key: awsSecretAccessKey
 ----
-<1> The Kafka Connect xref:type-KafkaConnectSpec-reference[version].
+<1> The Kafka Connect xref:type-KafkaConnectSpec-reference[version] used with the MirrorMaker 2.0 deployment, which must be from the same Kafka version as MirrorMaker 2.0.
```
Comment (scholzj) [inline documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc:183]: I think this should say something like _Defines the version of Kafka Connect and Mirror Maker 2 which will be used. They will be always the same_.

File: documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc (hunk 1 above)
Comment (PaulRMellor) [reply documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc:57]: removed

File: documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc (hunk 2 above)
Comment (PaulRMellor) [reply documentation/modules/configuring/proc-config-mirrormaker2-replication.adoc:183]: Updated to:

_The Kafka Connect and Mirror Maker 2.0 xref:type-KafkaConnectSpec-reference[version], which will always be the same._

**Review by ppatierno — APPROVED** [review]

**Review by scholzj — APPROVED** [review]

**Comment by scholzj** [comment]
@laidan6000 Could you please have a look at this?

**Review by laidan6000 — APPROVED** [review]
Looks good to me.

**Pull request merged**